contracts/types/GasEstimationTypes.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title GasEstimationType
+ * @notice This enum represents the gas estimation types for different chains.
+ */
+enum GasEstimationType {
+    Default,
+    OptimismBedrock,
+    OptimismEcotone,
+    Arbitrum,
+    Scroll,
+    Mantle
+}
+
+/**
+ * @title GasInfo
+ * @notice This struct represents the gas pricing information for a specific chain.
+ * @dev Using smaller uint types to save storage packed into 32 byte slots.
+ */
+struct GasInfo {
+    uint64 gasEstimationType; // Custom gas pricing rule, such as L1 data fee on L2s
```

Comment:
```suggestion
    /// @dev example
    uint64 gasEstimationType; // Custom gas pricing rule, such as L1 data fee on L2s
```
Maybe use this syntax, or detail it in the doc above the struct, so it's more readable?